### ISSUE TITLE
Our implementation does have back propagation and many other features

### ISSUE THREAD

**Issue by nleck** [stSoftwareAU/NEAT-AI-Examples#182]
This page seems to be comparing to standard NEAT algorithm https://github.com/stSoftwareAU/NEAT-AI-Examples/blob/Develop/mnist_classification/README.md

See https://github.com/stSoftwareAU/NEAT-AI/blob/Develop/COMPARISON.md I think you maybe quite surprised 

Our implementation has done a lot of work to improve the speed of evolution especially with large creatures/data which may not we obvious with our small examples

But even with MNIST classifier you can see we got to high success in a it appears a dozen generations or so ( looks like it) , if the training data is prepared in a binary format that would be every fast indeed

Please check all documentation against our implementation not just the simple NEAT algorithm, we have tried to bring in a broad range of approaches all aiming to improve evolution speed. Look at NEAT-AI-Discovery for example, this applies a bit of science to the structural changes in the network instead of just random mutations

**Comment by stservice** [comment]
Claimed by `stservice-1778281749485`

**Comment by stservice** [comment]
## 🗺️ Planning Summary

I read your specific concerns — `mnist_classification/README.md` reading as if NEAT-AI were just textbook NEAT, the missing acknowledgement of backpropagation and other features, the binary-data-format speed advantage, and NEAT-AI-Discovery applying *"a bit of science to the structural changes in the network instead of just random mutations"* — and broke the work into seven sub-issues. The first one is the foundation; the other six are the actual README rewrites and can run in parallel once the audit lands.

### Sub-issues created

| # | Title | Depends on |
| --- | --- | --- |
| #184 | Audit example READMEs against NEAT-AI `COMPARISON.md` feature list | — |
| #185 | Rewrite `mnist_classification` README to reflect NEAT-AI's full feature set | #184 |
| #186 | Add "Beyond Standard NEAT" feature breadth section to top-level `README.md` | #184 |
| #187 | Add "NEAT-AI Features Used" callout to every example README | #184 |
| #188 | Clarify "Vanilla NEAT" vs NEAT-AI in `synthetic_synapse` README | #184 |
| #189 | Reframe Discovery + Discovery-at-Scale READMEs around science-driven structural mutation | #184 |
| #190 | Document the binary `.bin` training stream speed advantage prominently | #184 |

### Suggested implementation order

1. **#184 first** — produces `docs/neat_ai_feature_audit.md` cross-referencing every README against upstream `COMPARISON.md`, so subsequent rewrites have a single source of truth and we don't fix the same thing twice.
2. **#185–#190 in parallel** — each focuses on one README or doc, fits comfortably in a single PR, and references the audit. They are independent of each other.

```mermaid
flowchart LR
    A[#184 Documentation audit]
    A --> B[#185 MNIST README rewrite]
    A --> C[#186 Top-level breadth section]
    A --> D[#187 Per-example feature callouts]
    A --> E[#188 Synthetic Synapse clarification]
    A --> F[#189 Discovery 'applies science' framing]
    A --> G[#190 Binary training stream doc]
```

### Assumptions and decisions

- I treated upstream [`NEAT-AI/COMPARISON.md`](https://github.com/stSoftwareAU/NEAT-AI/blob/Develop/COMPARISON.md) as the canonical feature list. Every sub-issue links to it rather than duplicating the content into this repo.
- I quoted your wording directly in #189 (*"a bit of science to the structural changes"*) and #190 (*"if the training data is prepared in a binary format that would be very fast indeed"*) so the rewrites stay faithful to the framing you asked for.
- I kept the existing "🌟 Unique Features Showcase" table on the top-level `README.md` and made #186 add a *complementary* breadth section above it, rather than reworking the table — minimises the risk of breaking the screenshot links.
- I did **not** propose any code changes to the example programs themselves; the issue is about documentation accuracy, so every sub-issue is README-only.
- Tests added by each sub-issue are "what" tests on the published Markdown contents, in line with `AGENTS.md`'s testing philosophy.

### Iteration

The `planning` label will be removed and this issue closed automatically once the worker picks up the plan. **Reopen this issue if you want to change the plan** — for example if you'd prefer a single mega-PR rather than seven focused ones, or if you want a specific feature (predictive coding, MCMC, hyperparameter self-adaptation) given its own dedicated docs page.

**Comment by stservice** [comment]
## Planning Complete

Planning complete. **7 sub-issue(s)** created:

- https://github.com/stSoftwareAU/NEAT-AI-Examples/issues/184
- https://github.com/stSoftwareAU/NEAT-AI-Examples/issues/185
- https://github.com/stSoftwareAU/NEAT-AI-Examples/issues/186
- https://github.com/stSoftwareAU/NEAT-AI-Examples/issues/187
- https://github.com/stSoftwareAU/NEAT-AI-Examples/issues/188
- https://github.com/stSoftwareAU/NEAT-AI-Examples/issues/189
- https://github.com/stSoftwareAU/NEAT-AI-Examples/issues/190

---
🤖 Processed by: stservice

**Comment by stservice** [comment]
Planning complete — sub-issues created.
